frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -28,15 +29,25 @@ export class SecretData extends React.PureComponent {
     this.setState({ showSecret: !this.state.showSecret });
   }
 
+  getValue(rawValue) {
+    if (_.isNil(rawValue)) {
+      return <span className="text-muted">No value</span>;
+    }
+
+    const { showSecret } = this.state;
+    const decodedValue = window.atob(rawValue);
+    const visibleValue = showSecret ? decodedValue : <MaskedData />;
+    return <CopyToClipboard value={decodedValue} visibleValue={visibleValue} />;
+  }
+
   render() {
     const { data } = this.props;
     const { showSecret } = this.state;
     const dl = [];
     Object.keys(data || {}).sort().forEach(k => {
```

Comment:
Looks good, though I still think we should use a `reduce()` here:

```JavaScript 
const dl = Object.keys(data || {}).sort().reduce((result, key) => {
  result.push( /* the markup */ )
  return result;
}, []);
```
but thats just a lingering nit 😄